pgmpy/inference/CausalInference.py

```diff
@@ -58,10 +56,17 @@ def __init__(self, model, set_nodes=None):
                 f"Causal Inference is only implemented for a model with variable names with string type. Found {bad_variable[0]} with type {bad_variable[1]}. Convert them to string to proceed."
             )
         self.model = model
-        self.set_nodes = _variable_or_iterable_to_set(set_nodes)
-        self.observed_variables = frozenset(self.model.nodes()).difference(
-            model.latents
-        )
+
+        if isinstance(self.model, BayesianNetwork):
+            self.observed_variables = frozenset(self.model.nodes()).difference(
+                model.latents
+            )
+            self.graph = self.model.to_directed()
+        else:
+            self.observed_variables = frozenset(self.model.observed)
+            self.graph = self.model.graph
```

Comment:
@Nuna7 I would suggest doing `self.graph = self.model.full_graph_struct`. The `self.graph` can then be treated as a standard DAG model, allowing us to use methods such as `DAG.active_trail_nodes` and `DAG.moralize` instead of reimplmenting them here.